Automattic for Agencies: Implement "Download Products" for the Marketplace

yashwin opened the pull request:
Resolves https://github.com/Automattic/jetpack-genesis/issues/303

## Proposed Changes

This PR implements the "Download Products" page for the A4A Marketplace.

NOTE: Please ignore the top steps, as we will fix it later

## Testing Instructions

- Open the Calyso live link for A4A.
- Follow the steps here to create the agency ID: D139529-code.
- Click the Marketplace menu item > Issue WooCommerce Extensions: Booking to test the download part. 
- Assign this license to a site > Verify you are redirected to the "Download Products" page, and it looks as shown below. Also try downloading the plugin and verify it gets downloaded.

<img width="1561" alt="Screenshot 2024-03-22 at 12 20 16 PM" src="https://github.com/Automattic/wp-calypso/assets/10586875/e0dca1a0-007a-4b98-afd0-56ddd8614037">

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

yashwin committed:
Implement Download Products page for A4A Marketplace

jkguidaven commented on client/sections.js:
```diff
@@ -735,6 +735,7 @@ const sections = [
 			'/marketplace/hosting',
 			'/marketplace/checkout',
 			'/marketplace/assign-license',
+			'marketplace/download-products',
```
[non-blocker] It has a missing slash in the beginning.

```suggestion
			'/marketplace/download-products',
```

jkguidaven commented on client/a8c-for-agencies/sections/marketplace/download-products-form/style.scss:
```diff
@@ -0,0 +1,110 @@
+.download-products-form__action-items pre {
+	font-size: 0.875rem;
+	padding: 0;
+	margin-bottom: 0.6rem;
```
This one can be improved using logical properties.

yashwin replied:
Sorry for missing this.

jkguidaven commented on client/a8c-for-agencies/sections/marketplace/download-products-form/style.scss:
```diff
@@ -0,0 +1,110 @@
+	align-self: flex-end;
+	margin: 0 0 1rem;
+	font-size: 0.875rem;
+	color: #333;
```
Is there a chance we can use an existing color variable for this one?

jkguidaven reviewed (approved):
As per test instruction, the changes work as expected. I left few non-blocking comments but the PR should be good to 🚢 👍

yashwin committed:
address PR comments

yashwin commented:
Thanks for the review, @jkguidaven. I have addressed your comments: https://github.com/Automattic/wp-calypso/pull/88787/commits/34fa665bbea8909679eac597533ae5000f78d53f
